crates/vision/src/line_detection.rs

```diff
@@ -234,6 +237,47 @@ fn get_gradient(image: &YCbCr422Image, point: Point2<u16>) -> Vector2<f32> {
         .unwrap_or_else(Vector2::zeros)
 }
 
+struct SegmentMerger<T: Iterator<Item = Segment>> {
+    iterator: Peekable<T>,
+    maximum_merge_gap: u16,
+}
+
+impl<T> SegmentMerger<T>
+where
+    T: Iterator<Item = Segment>,
+{
+    fn new(iterator: T, maximum_merge_gap: u16) -> Self {
+        Self {
+            iterator: iterator.peekable(),
+            maximum_merge_gap,
+        }
+    }
+}
+
+impl<T> Iterator for SegmentMerger<T>
+where
+    T: Iterator<Item = Segment>,
+{
+    type Item = Segment;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut current = self.iterator.next()?;
+
+        while let Some(next) = self.iterator.peek().copied() {
+            if next.start - current.end >= self.maximum_merge_gap {
+                break;
+            }
+
+            let _ = self.iterator.next();
```

Comment:
I think you should be able to remove `let _ =`

Reply:
Yes, but I think this is more readable. I explicitly and intentionally discard the return value here.